datastore/longhorn.go

```diff
@@ -304,6 +306,27 @@ func (s *DataStore) UpdateEngine(e *longhorn.Engine) (*longhorn.Engine, error) {
 	return s.lhClient.LonghornV1alpha1().Engines(s.namespace).Update(e)
 }
 
+func (s *DataStore) WaitEnginesDeleted(volumeName string) error {
```

Comment:
We don't want to block the reconciliation process. We can just check the existence of replicas and engines in the reconciliation process like before. If there is none, then the deletion was completed. Otherwise, we just continue and wait for the next loop. 

The volume controller is watching replicas and engines. The deletion of the object should trigger the volume controller reconciliation loop.